Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM golang:1.18 as builder
```

Comment:
Do we want to pin to a specific patch version here too (e.g. `golang:1.18.2`)? I could go either way but if we are sticking to just major/minor here, then we should update the build/test workflow to also specify only major/minor for consistency (https://github.com/actions/setup-go#supported-version-syntax)

Reply:
fixed !

Reply:
Should we also use a build stage for the binary and consider running a scratch or alpine container with the build binary? 

Something like this:

```
FROM golang:1.18 AS builder

WORKDIR /workspace
# Copy the Go Modules manifests
COPY go.mod go.mod
COPY go.sum go.sum
# cache deps before building and copying source so that we don't need to re-download as much
# and so that source changes don't invalidate our downloaded layer
RUN go mod download

# Copy the go source
COPY . /workspace

# Build
RUN make build
 
FROM scratch
COPY --from builder <binary-source> <dest> 
```

Reply:
we are doing something similar here, using `gcr.io/distroless/static:nonroot`. If this is too limiting, we could create another debug container based off `alpine` / `ubuntu` in the future

```
FROM gcr.io/distroless/static:nonroot
 WORKDIR /
 COPY --from=builder /workspace/bin/envoy-gateway .

 ```

Reply:
we are doing something similar here, using `gcr.io/distroless/static:nonroot`. If this is too limiting, we could create another debug container based off `alpine` / `ubuntu` in the future

```
FROM gcr.io/distroless/static:nonroot
 WORKDIR /
 COPY --from=builder /workspace/bin/envoy-gateway .

 ```

Reply:
I think the major issue I found using a minimal container was the lack of `ca-certificates` if connecting to an upstream that is not "internal".

Reply:
@jasonmccallister Distroless static includes [ca-certificates](https://github.com/GoogleContainerTools/distroless/blob/main/base/README.md). Since distroless is based off Debian, we could use  [update-ca-certificates](https://manpages.debian.org/buster/ca-certificates/update-ca-certificates.8.en.html) as part of a multi-stage build to include additional trusted CAs. For example:
```
FROM debian AS build-env

# Add CA certs
ADD my-ca-file.crt /usr/local/share/ca-certificates/my-ca-file.crt
RUN update-ca-certificates

FROM gcr.io/distroless/static
COPY --from=build-env /etc/ssl/certs /etc/ssl/certs
```
Please create an issue if this is a feature you need. PRs are always welcome ;-)

Reply:
That would work! I have not used the distress image but it sounds useful. We are essentially doing that with the upstream envoy image for our build. Not sure if its needed in the  final image, but might be useful to at least document?